base/loading.jl

```diff
@@ -671,20 +671,59 @@ function env_project_file(env::String)::Union{Bool,String}
 end
 
 function base_project(project_file)
-    base_dir = abspath(joinpath(dirname(project_file), ".."))
-    base_project_file = env_project_file(base_dir)
-    base_project_file isa String || return nothing
-    d = parsed_toml(base_project_file)
-    workspace = get(d, "workspace", nothing)::Union{Dict{String, Any}, Nothing}
-    if workspace === nothing
-        return nothing
-    end
-    projects = get(workspace, "projects", nothing)::Union{Vector{String}, Nothing, String}
-    projects === nothing && return nothing
-    if projects isa Vector && basename(dirname(project_file)) in projects
-        return base_project_file
+    home_dir = abspath(homedir())
+    project_dir = abspath(dirname(project_file))
+    current_dir = project_dir
+
+    debug = get(ENV, "JULIA_DEBUG_LOADING_WORKSPACE", nothing) == "1"
+    debug && println("DEBUG: base_project called with project_file: ", project_file)
+    debug && println("DEBUG: home_dir: ", home_dir)
+    debug && println("DEBUG: project_dir: ", project_dir)
+
+    while true
+        parent_dir = abspath(joinpath(current_dir, ".."))
```

Comment:
`abspath joinpath ..` is not a string operation, and will not terminate this loop correctly. you must use dirname